setup.cfg

```diff
@@ -9,3 +9,15 @@ exclude_lines =
     if __name__ == .__main__.:
 
     self.logger.debug
+
+[flake8]
+max-line-length = 88
+max-complexity = 8
```

Comment:
I'm noticing that `Parser.parse_csv` and `Lexer.scan` are both exceeding this complexity score when I run the test suite locally. I'm not sure if that's something that can really be fixed as the branching logic is a necessary part of how a recursive descent parser works. What's your opinion on this aspect of linting? Is it something that could be disabled?

Reply:
Good question. In cases like this, I generally prefer to leave the cyclomatic complexity check enabled globally and skip w/ a `# noqa: C901` in places where there was a conscious decision to have very complex statements.

As one example, I identified the opportunity to consolidate the logic in the [`scan` method](https://github.com/joshtemple/lkml/pull/19#discussion-diff-311616621R86) because of this check.

Reply:
Great, do you mind adding that comment to those two methods? Then LGTM!

Reply:
One point of clarification--could you share the specific failures you're seeing locally? The test suite is passing for me both locally & in [CircleCI](https://circleci.com/gh/jamescurtin/lkml/12?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link). Want to make sure I'm not missing something before I add those ignores.

Reply:
When I run `docker-compose run --rm test`, I get the following result for `flake8` and a failing exit code.

```
Running flake8...
./build/lib/lkml/parser.py:326:5: C901 'Parser.parse_csv' is too complex (10)
./build/lib/lkml/lexer.py:59:5: C901 'Lexer.scan' is too complex (13)
```

Reply:
Looks like those files are old versions (`scan` doesn't have your changes). Any idea how that `build` directory gets generated and how to refresh it?

Reply:
Never mind, it's a `build` directory that I have that you don't, probably generated by `twine` or something. I think we just need to change `flake8` to `flake8 lkml` to make sure we're only linting the source code.

Reply:
Ah, makes sense! Just pushed the final change. Should be good to go!